modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/PlayerHeaderFragment.kt

```diff
@@ -579,14 +579,30 @@ class PlayerHeaderFragment : BaseFragment(), PlayerClickListener {
             return
         }
 
+        val isExistingBookmark = isExist(result.bookmarkUuid)
+
+        val snackbarMessage = if (isExistingBookmark) {
+            getString(LR.string.bookmark_exists, result.title)
+        } else {
+            getString(LR.string.bookmark_added, result.title)
+        }
         val viewBookmarksAction = View.OnClickListener {
             (parentFragment as? PlayerContainerFragment)?.openBookmarks()
         }
-        Snackbar.make(view, getString(LR.string.bookmark_added, result.title), Snackbar.LENGTH_LONG)
+
+        val existingBookmarkUuid = result.bookmarkUuid
+        settings.setExistingBookmarkUuid(existingBookmarkUuid)
```

Comment:
There's a problem with this approach. If we add a bookmark at the first bookmark position after creating a second bookmark at another position, my first bookmark uuid will not be counted as an existing one since the second one has replaced it.

See if this video helps:

https://github.com/Automattic/pocket-casts-android/assets/1405144/447d6c2c-8d19-4b37-817e-1e301dfbe64e

We can see if passing `isExistingBookmark` from the `BookmarkViewModel`'s `saveBookmark(...)` down to  `BookmarkActivityContract.BookmarkResult` just like other params `bookmarkUuid`, `title`, `tintColor` helps.

Note that there's also a `showViewBookmarksSnackbar` on `MainActivity`.

Reply:
> There's a problem with this approach. If we add a bookmark at the first bookmark position after creating a second bookmark at another position, my first bookmark uuid will not be counted as an existing one since the second one has replaced it.
> 
> See if this video helps:
> 
>  Screen_recording_20230825_161002.mp4 
> We can see if passing `isExistingBookmark` from the `BookmarkViewModel`'s `saveBookmark(...)` down to `BookmarkActivityContract.BookmarkResult` just like other params `bookmarkUuid`, `title`, `tintColor` helps.
> 
> Note that there's also a `showViewBookmarksSnackbar` on `MainActivity`.

mmh interesting. I didn't notice that. Let me try passing the `isExistingBookmark` flag from the viewmodel.